### ISSUE TITLE
Intermittent failures when calling sast wait-for

### ISSUE THREAD

**Issue by anglind** [fortify/fcli#500]
We are getting intermittent errors when calling `fod sast wait-for`. 
We are first using `fod sast start` to upload and start a new scan but when we try to wait for that scan to complete, it intermittently fails saying it cannot find the scan.
Example failure:
```
10:42:22  + fcli fod sast start -f=/home/jenkins/agent/workspace/app@tmp/fodupload.zip --notes=https://github.com/xxx/xxxx.git --entitlement-preference=SubscriptionOnly --remediation-preference=NonRemediationScanOnly --in-progress-action=Queue 1020996
10:42:33  
Upload fodupload.zip: 0 of 97578960 bytes complete
10:42:33  
...  
10:43:25  
Upload fodupload.zip: 97517568 of 97578960 bytes complete
10:43:25  
10:43:25   Id       Assessment Type  Analysis Status  Application  Microservice  Release  Started  Completed  Scan Method  Action  
10:43:25   9509262  N/A              Pending          N/A          N/A           N/A      N/A      N/A        N/A          STARTED
10:43:25  
10:43:25  [Pipeline] timeout
10:43:25  Timeout set to expire in 30 min
10:43:25  [Pipeline] {
10:43:25  [Pipeline] isUnix
10:43:25  [Pipeline] sh
10:43:26  + fcli fod release get 1020996 -o json
10:43:27  [Pipeline] readJSON
10:43:27  [Pipeline] isUnix
10:43:27  [Pipeline] sh
10:43:28  + fcli fod sast wait-for --interval=1m 9509262
10:43:29  com.fortify.cli.fod.scan.helper.FoDScanNotFoundException: Could not retrieve scan with id: 9509262
10:43:29   at com.fortify.cli.fod.scan.helper.FoDScanHelper.getScanDescriptor(FoDScanHelper.java:153)
10:43:29   at com.fortify.cli.fod.scan.cli.mixin.FoDScanResolverMixin$AbstractFoDMultiScanResolverMixin.lambda$getScanDescriptors$0(FoDScanResolverMixin.java:41)
```

But sometimes it will work just fine:
```
06:40:17  + fcli fod sast start -f=/home/jenkins/agent/workspace/app@tmp/fodupload.zip --notes=https://github.com/xxx/xxxx.git --entitlement-preference=SubscriptionOnly --remediation-preference=NonRemediationScanOnly --in-progress-action=Queue 1020996
06:40:30  
Upload fodupload.zip: 0 of 97579126 bytes complete
06:40:30  
...
06:41:26  
Upload fodupload.zip: 97517568 of 97579126 bytes complete
06:41:26  
06:41:26   Id       Assessment Type  Analysis Status  Application  Microservice  Release  Started  Completed  Scan Method  Action  
06:41:26   9520658  N/A              Pending          N/A          N/A           N/A      N/A      N/A        N/A          STARTED
06:41:26  
06:41:27  [Pipeline] timeout
06:41:27  Timeout set to expire in 30 min
06:41:27  [Pipeline] {
06:41:27  [Pipeline] isUnix
06:41:27  [Pipeline] sh
06:41:27  + fcli fod release get 1020996 -o json
06:41:28  [Pipeline] readJSON
06:41:29  [Pipeline] isUnix
06:41:29  [Pipeline] sh
06:41:29  + fcli fod sast wait-for --interval=1m 9520658
06:41:30   Id       Assessment Type    Analysis Status  Application Microservice  Release      Started              Completed  Scan Method  Action  
06:41:30   9520658  Static Assessment  Queued           App   		N/A           Development  2024-01-11T06:41:25  N/A        Other        WAITING
06:41:30  
06:42:40   Id       Assessment Type    Analysis Status  Application Microservice  Release      Started              Completed  Scan Method  Action  
06:42:40   9520658  Static Assessment  In_Progress      App   		N/A           Development  2024-01-11T06:41:47  N/A        Other        WAITING
06:42:40  
06:43:39   Id       Assessment Type    Analysis Status  Application Microservice  Release      Started              Completed  Scan Method  Action  
06:43:39   9520658  Static Assessment  In_Progress      App   		N/A           Development  2024-01-11T06:41:47  N/A        Other        WAITING
06:43:39  
06:44:37   Id       Assessment Type    Analysis Status  Application Microservice  Release      Started              Completed  Scan Method  Action  
06:44:37   9520658  Static Assessment  In_Progress      App   		N/A           Development  2024-01-11T06:41:47  N/A        Other        WAITING
06:44:37  
06:45:36   Id       Assessment Type    Analysis Status  Application Microservice  Release      Started              Completed            Scan Method  Action        
06:45:36   9520658  Static Assessment  Completed        App   		N/A           Development  2024-01-11T06:41:47  2024-01-11T06:44:12  Other        WAIT_COMPLETE
06:45:36  
06:45:36   Id       Assessment Type    Analysis Status  Application Microservice  Release      Started              Completed            Scan Method  Action        
06:45:36   9520658  Static Assessment  Completed        App   		N/A           Development  2024-01-11T06:41:47  2024-01-11T06:44:12  Other        WAIT_COMPLETE 
```
We are using version 1.3.0.

**Comment by rsenden** [comment]
Thanks for reporting. This is probably an issue on the FoD-side, where it may take a little time before a newly imported or submitted scan is accessible through the REST endpoints. I think this was already reported this to the FoD team before as we occasionally experienced similar issues during fcli functional tests, but I'll follow up on that. We'll also think about implementing a work-around in fcli.

For now, probably best to simply add a delay of a couple of seconds between the `fcli fod sast start` and `fcli fod sast wait-for` commands.

Side note; fcli 1.3.0 is a somewhat older version, latest version at the moment is 2.1.0, with a lot of improvements and bug fixes (together with many consistency changes to fcli command structure, so you may need to adjust your existing fcli invocations). I don't think we explicitly addressed this particular issue, but you could give 2.1.0 a try to see whether this exhibits the same behavior.

**Comment by anglind** [comment]
Thanks. Yeah I also assume this is more of a platform issue as opposed to an issue with the CLI. I talked to Fortify support and they directed me to open this issue here.

We'll have a look at consuming the latest version and adding a delay as a quick workaround.

**Comment by rsenden** [comment]
This issue has been fixed in fcli 2.4.0
